packages/fuel-indexer-database/postgres/src/lib.rs

```diff
@@ -757,40 +757,113 @@ pub async fn revert_transaction(
 }
 
 #[cfg_attr(feature = "metrics", metrics)]
-pub async fn remove_indexer(
+pub async fn remove_graph(
     conn: &mut PoolConnection<Postgres>,
     namespace: &str,
     identifier: &str,
 ) -> sqlx::Result<()> {
-    let index_id = get_indexer_id(conn, namespace, identifier).await?;
+    execute_query(
+        conn,
+        format!(
+            "DELETE FROM graph_registry_columns WHERE type_id IN (SELECT id FROM graph_registry_type_ids WHERE schema_name = '{namespace}' AND schema_identifier = '{identifier}');"
+        ),
+    )
+    .await?;
 
     execute_query(
         conn,
-        format!("DELETE FROM index_asset_registry_wasm WHERE index_id = {index_id}"),
+        format!(
+            "DELETE FROM graph_registry_type_ids WHERE schema_name = '{namespace}' AND schema_identifier = '{identifier}';"
+        ),
     )
     .await?;
 
     execute_query(
         conn,
-        format!("DELETE FROM index_asset_registry_manifest WHERE index_id = {index_id}"),
+        format!(
+            "DELETE FROM graph_registry_root_columns WHERE root_id = (SELECT id FROM graph_registry_graph_root WHERE schema_name = '{namespace}' AND schema_identifier = '{identifier}');"
+        ),
     )
     .await?;
 
     execute_query(
         conn,
-        format!("DELETE FROM index_asset_registry_schema WHERE index_id = {index_id}"),
+        format!(
+            "DELETE FROM graph_registry_graph_root WHERE schema_name = '{namespace}' AND schema_identifier = '{identifier}';"
+        ),
     )
     .await?;
 
     execute_query(
         conn,
-        format!("DELETE FROM index_registry WHERE id = {index_id}"),
+        format!("DROP SCHEMA {namespace}_{identifier} CASCADE"),
     )
     .await?;
 
     Ok(())
 }
 
+#[cfg_attr(feature = "metrics", metrics)]
+pub async fn remove_indexers(
+    conn: &mut PoolConnection<Postgres>,
+    namespace: &str,
+    identifier: &str,
+    limit: Option<usize>,
+) -> sqlx::Result<()> {
+    let limit = limit.map(|l| format!("LIMIT {l}")).unwrap_or_default();
```

Comment:
Due to the foreign key constraints we're using, there cannot be an indexer with the same namespace and identifier, so I think we can remove this?